express-notes-api/server.js

```diff
@@ -0,0 +1,171 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  const notesArray = [];
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+    for (const key in notes) {
+      notesArray.push(notes[key]);
+    }
+    if (notes) {
+      res.json(notesArray);
+    }
+  } catch (error) {
+    res.json([]);
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const errorResponse = {};
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+
+    if (parseInt(req.params.id) < 0) {
+      errorResponse.error = 'id must be a positive integer';
+      throw new Error();
+    }
+    if (notes[parseInt(req.params.id)] === undefined) {
+      errorResponse.error = `cannot find note with id ${req.params.id}`;
+      throw new Error();
+    }
+
+    res.status(200).json(notes[parseInt(req.params.id)]);
+    return;
+  } catch (error) {
+    switch (errorResponse.error) {
+      case 'id must be a positive integer':
+        res.status(400).send(errorResponse);
+        break;
+      case `cannot find note with id ${req.params.id}`:
+        res.status(404).send(errorResponse);
+        break;
+      default:
+        errorResponse.error = 'an unexpected error has occured';
+        res.status(500).send(errorResponse);
+        break;
+    }
+  }
+});
+
+app.post('/api/notes', async (req, res) => {
+  const errorResponse = {};
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+
+    if (req.body.content === undefined) {
```

Comment:
I don't like direct comparisons to `undefined`, though sometimes it's necessary. You could instead use `if (!req.body.content)`. That has the advantage that if the content is present but an empty string, it will also trigger the error.